core/deployment/src/main/java/io/quarkus/deployment/dev/RuntimeUpdatesProcessor.java

```diff
@@ -222,11 +217,12 @@ public boolean doScan(boolean userInitiated) throws IOException {
                     }
                     Index current = indexer.complete();
                     boolean ok = instrumentationEnabled()
-                            && !containsStartupCode(current);
+                            && !disableInstrumentationForIndexPredicate.test(current);
```

Comment:
Do we really want to replace the containsStartupCode here? It looks like start up code is an exception anyway.

Reply:
Ah, sorry, you moved it.

Reply:
Exactly :)